src/wbase/Task.h

```diff
@@ -220,16 +220,41 @@ class Task : public util::CommandForThreadPool {
     /// Return a json object describing sdome details of this task.
     nlohmann::json getJson() const;
 
+    int64_t getSession() const { return _session; }
+    int getProtocol() const { return _protocol; }
+    std::string getDb() const { return _db; }
+    int getCzarId() const { return _czarId; }
+    bool getFragmentHasSubchunks() const { return _fragmentHasSubchunks; }
+    int getSubchunkId() const { return _subchunkId; }
+
+    /// Do not alter the returned value. &&& ??? just return a copy?
+    DbTableSet& getDbTbls() { return _dbTbls; }
+
+    /// Do not alter the returned value. &&& ??? just return a copy?
+    IntVector& getSubchunksVect() { return _subchunksVect; }
+
 private:
+    std::shared_ptr<UserQueryInfo> _userQueryInfo;  ///< Details common to Tasks in this UserQuery.
     std::shared_ptr<SendChannelShared> _sendChannel;
     uint64_t const _tSeq = 0;     ///< identifier for the specific task
     QueryId const _qId = 0;       ///< queryId from czar
+    size_t const _templateId;     ///< Id number of the template in _userQueryInfo.
+    bool const _hasChunkId;       ///< True if there was a chunkId in the czar message.
+    int const _chunkId;           ///< chunkId from czar
+    int const _subchunkId;        ///< subchunkId from czar
     int const _jId = 0;           ///< jobId from czar
     int const _attemptCount = 0;  // attemptCount from czar
-    /// _idStr for logging only.
-    std::string const _idStr = makeIdStr(true);
-    std::string _queryString;   ///< The query this task will run.
-    int _queryFragmentNum = 0;  ///< The fragment number of the query in the task message.
+    int const _queryFragmentNum;  ///< The fragment number of the query in the task message. // &&& delete???
+    bool const _fragmentHasSubchunks;  ///< True if the fragment in this query has subchunks.
+    int64_t const _session;            ///< XrdSsi session.
+    bool const _hasDb;                 ///< true if db was in message from czar.
+    std::string _db;                   ///< Task database (&&& move to UserQueryInfo ???)
```

Comment:
Why this is not declared as `const`? Is the value of this attribute going to be modified later?